### PR TITLE
Adds an Actions generator to the cmd package

### PR DESCRIPTION
**Pull request by apaganobeleno**
@markbates this covers the basic functionality described in #16 it still needs to cover some cases like when the tests signature already exist and the same for the templates, i thought this is a good place to start, i would like to get your feedback.

### CODE REVIEW

File: buffalo/cmd/generate/action.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+		data := gentronics.Data{"under": inflect.Underscore(name)}
+		_, err := os.Stat(filepath.Join("actions", fmt.Sprintf("%v.go", data["under"])))
+		fileExists := err == nil
+
+		g := gentronics.New()
+
+		if !fileExists {
+
+			g.Add(gentronics.NewFile(filepath.Join("actions", fmt.Sprintf("%s.go", data["under"])), rActionFileT))
+			g.Add(gentronics.NewFile(filepath.Join("actions", fmt.Sprintf("%s_test.go", data["under"])), rActionTest))
```
Comment (markbates) [inline buffalo/cmd/generate/action.go]: you need to add the template html file as well.

File: buffalo/cmd/generate/action.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	return aa
+}
+
+type actionAppender struct {
+	Namespace  string
+	ActionName string
+	FileName   string
+}
+
+func (aa actionAppender) Run(rootPath string, data gentronics.Data) error {
```
Comment (markbates) [inline buffalo/cmd/generate/action.go]: you should add all of this to the gentronics generator.

File: buffalo/cmd/generate/action.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+const (
+	rActionFileT = `package actions
+    import "github.com/gobuffalo/buffalo"`
+
+	rViewT       = `<h1>{{namespace}}#{{action}}</h1>`
+	rActionFuncT = `
+    
+    // {{namespace}}{{action}} default implementation.
+    func {{namespace}}{{action}}(c buffalo.Context) error {
+	    return c.Render(200, r.String("OK"))
```
Comment (markbates) [inline buffalo/cmd/generate/action.go]: this should be `r.HTML("{{namespace}}/{{action}}.html")`

File: buffalo/cmd/generate/action.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
+// Copyright © 2016 NAME HERE <EMAIL ADDRESS>
```
Comment (briandowns) [inline buffalo/cmd/generate/action.go]: Should this be updated before merge?

Comment (markbates) [inline buffalo/cmd/generate/action.go]: yes please. this isn't quite ready to merge yet.

Comment (apaganobeleno) [inline buffalo/cmd/generate/action.go]: @markbates @briandowns what should it say ?

File: buffalo/cmd/generate/action.go (hunk 3 above)
Comment (apaganobeleno) [inline buffalo/cmd/generate/action.go]: thanks for this one! i missed it.

File: buffalo/cmd/generate/action.go (hunk 4 above)
Comment (markbates) [inline buffalo/cmd/generate/action.go]: @apaganobeleno whatever is at the top of the other ones? or delete it. don't really care either way. you know what, just delete it. we have a license covering the project. don't need another one there.

File: buffalo/cmd/generate/action.go (hunk 2 above)
Comment (apaganobeleno) [inline buffalo/cmd/generate/action.go]: I moved this to a `gentronics.Runnable` that will append to existing file, is that ok ? is there a gentronic default runnable that appends to a file ?

File: buffalo/cmd/generate/action.go (hunk 1 above)
Comment (apaganobeleno) [inline buffalo/cmd/generate/action.go]: @markbates i had to work the actions separately given for the same file will be multiple actions, did i get it correctly ? 🤔

File: buffalo/cmd/generate/action.go (hunk 2 above)
Comment (markbates) [inline buffalo/cmd/generate/action.go]: no, i guess there isn't. :( but that should be awesome to have! :wink: :wink:

**Comment by markbates** [comment]
@apaganobeleno wrong `inflect` package. :)

```
buffalo/cmd/generate/action.go:12:2: cannot find package "github.com/bep/inflect" in any of:
	/usr/local/go/src/github.com/bep/inflect (from $GOROOT)
	/Users/markbates/Dropbox/development/gocode/src/github.com/bep/inflect (from $GOPATH)
```

should be `github.com/markbates/inflect`

**Comment by apaganobeleno** [comment]
@markbates ah! good catch!

**Comment by markbates** [comment]
@apaganobeleno can you print out the names of the templates that are being generated? keep it uniformed with the rest of the output. You could move those to a generator instead of doing it manually, that will also fix it.

After that, I think we're good to go! :)

**Comment by apaganobeleno** [comment]
@markbates all set on printing the templates. is this how you would like it to look ?
```
Buffalo version 0.7.0

--> actions/cars.go
--> actions/cars_test.go
--> actions/cars.go
--> actions/cars_test.go
--> templates/cars/show.html
--> templates/cars/list.html
--> goimports -w .
```

**Comment by apaganobeleno** [comment]
@markbates now that i gave some thought to what you mentioned i could have built the template to be passed to the generator instead writing my own generator (#needABenderIconHere) i can work on that as a refactoring task if you consider its needed, for now i agree we're good to go.

**Comment by markbates** [comment]
Merged!! Woot! Thank you Antonio!

**Comment by apaganobeleno** [comment]
thanks @markbates i was doing some refactoring and ended up with something, would you like to take a look? i went trough your suggestion of using gentronics and ended up with something simpler, thanks for that, sending another PR for that, it passes the same set of tests.

**Comment by markbates** [comment]
@apaganobeleno yeah, i would love to see something simpler. :) always a fan of simpler.

**Pull request merged**